drivers/oled/oled_driver_user_task.c

```diff
@@ -0,0 +1,3 @@
+#include "oled_driver.h"
+
+void oled_task_user(void) { }
```

Comment:
As this is always added to `QUANTUM_LIB_SRC`, this wont work without it being declared weak?

Reply:
```makefile
ifeq ($(strip $(OLED_DRIVER_ENABLE)), yes)
    OPT_DEFS += -DOLED_DRIVER_ENABLE
    COMMON_VPATH += $(DRIVER_PATH)/oled
    QUANTUM_LIB_SRC += i2c_master.c
    QUANTUM_LIB_SRC += oled_driver_user_task.c
    SRC += oled_driver.c
endif
```

If `oled_task_user()` is defined in the keyboard or keymap code, `oled_driver_user_task.o` will not be linked. That is the function of QUANTUM_LIB_SRC and LIB_SRC.

Reply:
Given we use weak for virtually every single point of customisation, i really dont see the need to fragment with a different way to do things.

Reply:
There is no change in the existing customization methods for users. It will still be overridden if the user defines oled_task_user().

This PR also provides a way to override the default oled_task_user() and other functions in a keyboard-level file with a keymap-level implementation of oled_task_user().

Reply:
There is no change in the existing customization methods for users. It will still be overridden if the user defines oled_task_user().

This PR also provides a way to override the default oled_task_user() and other functions in a keyboard-level file with a keymap-level implementation of oled_task_user().